src/you_get/extractors/bilibili.py

```diff
@@ -474,9 +570,66 @@ def download_playlist_by_url(self, url, **kwargs):
             initial_state = json.loads(initial_state_text)
             aid = initial_state['videoData']['aid']
             pn = initial_state['videoData']['videos']
-            for pi in range(1, pn + 1):
-                purl = 'https://www.bilibili.com/video/av%s?p=%s' % (aid, pi)
-                self.__class__().download_by_url(purl, **kwargs)
+            if pn!= len(initial_state['videoData']['pages']):#interaction video 互动视频
+                search_node_list = []
+                download_cid_set = set([initial_state['videoData']['cid']])
+                params = {
+                        'id': 'cid:{}'.format(initial_state['videoData']['cid']),
+                        'aid': str(aid)
+                }
+                urlcontent = get_content('https://api.bilibili.com/x/player.so?'+parse.urlencode(params), headers=self.bilibili_headers(referer='https://www.bilibili.com/video/av{}'.format(aid)))
+                graph_version = json.loads(urlcontent[urlcontent.find('<interaction>')+13:urlcontent.find('</interaction>')])['graph_version']
+                params = {
+                    'aid': str(aid),
+                    'graph_version': graph_version,
+                    'platform': 'pc',
+                    'portal': 0,
+                    'screen': 0,
+                }
+                node_info = json.loads(get_content('https://api.bilibili.com/x/stein/nodeinfo?'+parse.urlencode(params)))
+
+                playinfo_text = match1(html_content, r'__playinfo__=(.*?)</script><script>')  # FIXME
+                playinfo = json.loads(playinfo_text) if playinfo_text else None
+
+                html_content_ = get_content(self.url, headers=self.bilibili_headers(cookie='CURRENT_FNVAL=16'))
+                playinfo_text_ = match1(html_content_, r'__playinfo__=(.*?)</script><script>')  # FIXME
+                playinfo_ = json.loads(playinfo_text_) if playinfo_text_ else None
```

Comment:
Reusing `playinfo` and `playinfo_` causes a serious bug for regular multi-part videos: if the 2nd video does not have the same DASH stream quality as the 1st one, `you-get -l` will end up downloading the same video (1st one) for all parts.
 
Examples:
- you-get -l https://www.bilibili.com/video/BV1fc41187xA  
- you-get -l https://www.bilibili.com/video/BV1ms411J7V3

You get the same video for all parts.

This has been a longstanding issue I haven't noticed before; Interaction videos should be handled separately and not to be confused with regular multi-part videos.